docs/concepts/gas.md

```diff
@@ -4,10 +4,159 @@ title: Gas
+
+### Function calls
+
+Given the general-purpose nature of NEAR, function calls win the award for most complex gas calculations. A given function call will use a hard-to-predict amount of CPU, network, and IO, and the amount of each can even change based on the amount of data already stored in the contract!
+
+With this level of complexity, it's no longer useful to walk through an example, [enumerating each](https://github.com/nearprotocol/nearcore/blob/f816d09ad634071aff20ad1c71aaf0f6886742d5/neard/res/genesis_config.json#L135-L185) of the gas calculations as we go (you can research this yourself, [if you want](https://github.com/nearprotocol/nearcore/pull/3038)). Instead, let's approach this from two other angles: ballpark comparisons to Ethereum, and getting accurate estimates with automated tests.
+
+#### Ballpark Comparisons to Ethereum
+
+Like NEAR, Ethereum uses gas units to model computational complexity of an operation. Unlike NEAR, rather than using a predictable gas price, Ethereum uses a dynamic, auction-based marketplace. This makes a comparison to Ethereum's gas prices a little tricky, but we'll do our best.
+
+Etherscan gives a [historic Ethereum gas price chart](https://etherscan.io/chart/gasprice). These prices are given in "Gwei", or Gigawei, where a wei is the smallest possible amount of ETH, 10<super>-18</super>. From November 2017 through July 2020, average gas price was 21Gwei. Let's call this the "average" gas price. In July 2020, average gas price went up to 57Gwei. Let's use this as a "high" Ethereum gas fee.
+
+Multiplying Ethereum's gas units by gas price usually results in an amount that easy to show in milliETH (mE), the same way we've been converting NEAR's TGas to milliNEAR. Let's look at some common operations side-by-side, comparing ETH's gas units to NEAR's, as well as converting to both the above "average" & "high" gas prices.
+
+| Operation                                       | ETH gas units | avg mE | high mE | NEAR TGas            | mN
+| ----------------------------------------------- | ------------- | ------ | ------- | ---------            | -----
+| Transfer native token (ETH or NEAR)             | 21k           |  0.441 |  1.197  |   0.45               | 0.045
+| Deploy & initialize a [fungible token] contract | [1.1M]        | 23.3   | 63.1    |   0.81               | 0.081 (plus 1.5Ⓝ in [storage staking])
+| Transfer a fungible token                       | [~45k]        |  0.945 |  2.565  | [14]<super>†</super> | 1.4
+| Setting an escrow for a fungible token          | [44k]         |  0.926 |  2.51   |  [8]                 | 0.8
+| Checking a balance for a fungible token         | 0             |  0     |  0      |   0                  | 0
+
+<super>†</super> Function calls require spinning up a VM and loading all compiled Wasm bytes into memory, hence the increased cost over base operations; this is [being optimized](???).
```

Comment:
@nearmax is there a an existing issue for tracking runtime-starting/Wasm-loading optimization?

Reply:
https://github.com/nearprotocol/nearcore/issues/3094